content/docs/user-guide/scripting/script-canvas/programmer-guide/behavior-context.md

```diff
@@ -4,91 +4,241 @@ description: Learn about the important relationship between Script Canvas and th
+{
+    if (m_light.IsOn())
+    {
+        TurnOffLight();
+    }
+    else
+    {
+        TurnOnLight();
+    }
+}
+```
 
-The following code shows the binding for the `BehaviorLightComponentNotificationBusHandler` and defines two events: `LightTurnedOn` and `LightTurnedOff`.
+To make these events accessible for scripting, their methods must be reflected to the behavior context. This is done in `LightComponent::Reflect`.
+
+```cpp
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorContext->EBus<LightComponentRequestBus>("Light", "LightComponentRequestBus")
+        ->Attribute(AZ::Script::Attributes::Category, "Rendering")
+        ->Event("TurnOn", &LightComponentRequestBus::Events::TurnOnLight)
+        ->Event("TurnOff", &LightComponentRequestBus::Events::TurnOffLight)
+        ->Event("Toggle", &LightComponentRequestBus::Events::ToggleLight);
+}
+```
+
+When Script Canvas examines the behavior context, it finds these events and automatically generates the corresponding nodes.
+
+![Light component nodes in Script Canvas](/images/user-guide/scripting/script-canvas/behavior-context-ebus-request-light-nodes.png)
+
+#### EBus events and EntityID
+
+EBuses communicate with an entity's components. To do this, it needs an address. All component EBuses derive from `AZ::ComponentBus`, which is addressable by an ID of the type `AZ::EntityId`. For this reason, all nodes from a component EBus have a data pin for an `EntityID`. The presence of `Self` in the `EntityID` field refers to the `EntityID` of the entity that owns the Script Canvas graph. However, this ID can be assigned to another entity, or even changed to an invalid entity ID.
+
+![Self EntityID](/images/user-guide/scripting/script-canvas/behavior-context-ebus-request-entityID.png)
+
+#### Tooltips
+
+Script Canvas nodes should include a helpful tooltip for every parameter. For example, the Light component might have a `SetLightState` event with a `state` parameter:
+
+```cpp
+// Set the light state to on or off.
+void SetLightState(State state);
+```
+
+You should add a tooltip in the behavior context reflection to describe the parameter. In this example, the tooltip will show "1=On, 0=Off" when a user hovers over the `State` parameter on the `SetState` node.
+
+```cpp
+    behaviorContext->EBus<LightComponentRequestBus>("Light", "LightComponentRequestBus")
+        ->Event("SetState", &LightComponentRequestBus::Events::SetLightState, {{{"State", "1=On, 0=Off"}}});
+```
+
+### Notification bus
+
+A notification bus is an EBus that enables the use of *event handlers*. Event handlers in a component can respond to the events that are sent to the component. You can reflect event handlers to the behavior context to make them available to scripting environments such as Script Canvas.
+
+Our Light component example handles the TurnOn, TurnOff, and Toggle events using the following C++ event handler methods on the `LightComponentNotificationBus`:
```

Comment:
```suggestion
Our Light component example handles the `LightTurnedOn` and `LightTurnedOff` events using the following C++ event handler methods on the `LightComponentNotificationBus`:
```

Reply:
I think I understand what you're getting at in this suggestion, but I'm going to leave this as is. Reason being: I wanted to refer to the original bound event, to show how TurnOn (request bus event) is handled by the following LightTurnedOn (notification bus event) event handler, for example.

Reply:
I'll add some clarification, though.